### ISSUE TITLE
Adding detailed migration instructions to the wiki

### ISSUE THREAD

**Issue by ravibanks54** [onnela-lab/beiwe-backend#111]
Hello,

I've set up a beiwe-backend cluster on AWS and everything seems to be working great so far. However, I've added new android sensors (specifically light, ambient temperature and pressure) to the frontend and want to support them in the backend (just enablement status and on/off settings). I've added the new data-stream settings to the study models but I'm having issues creating new studies.

I'm not super familiar with Flask/Django but I understand that the backing database needs to be updated when making model changes. When I run "python manage.py makemigrations" to create the new migration file, I'm running into a couple issues:

1. config/__init__.py can't find mandatory environment variables (I'm setting these from AWS). I verified from hopping into the interpreter that os.environ is set with all the appropriate variables but I must be missing something. I can brute-force fix this by updating manage.py to set the appropriate environment variables.

2. After I 'fix' the first issue, I'm getting an import error when running makemigrations:
sudo python manage.py makemigrations
Traceback (most recent call last):
  File "manage.py", line 25, in <module>
    from django.core.management import execute_from_command_line
ImportError: No module named django.core.management

How are migrations normally performed when you guys update the models? Am I missing something? It would be awesome if you could create a wiki guide on how to do this

**Comment by ravibanks54** [comment]
I was able to get it working by 
1. Not using sudo and
2. Changing the permissions of database/migrations to make it writable by my user.

Instructions:
1. Run 'python manage.py makemigrations'
2. Confirm schema changes
3. Run 'python manage.py migrate'

**Comment by zagorsky** [comment]
@ravibanks54 I'm glad you figured that out.  If you want to open pull requests on beiwe-android and beiwe-backend to add those new Android sensors, we'd be very happy to test and merge them.

**Comment by biblicabeebli** [comment]
I'm aware this is already closed, just commenting to say that it sounds like you are on an elastic beanstalk server, not a data processing server.  EB servers set ownership of the project folder to root.  the ubuntu processing servers are all owned by the ubuntu user.

Migrations run as part of EB deploy operations, if they fail then the whole deploy fails.  I'm not sure what you did to get into the situation you found yourself in.  If you deployed while mid-ssh'd into the `current` folder it may have been removed out from under you, causing weird things to happen.

**Comment by ravibanks54** [comment]
Yep, I have an EB server spun up. Part of my confusion is likely due to my lack of knowledge of django and database migrations. I'm assuming the part where I went wrong was that I should've run 'python manage.py makemigrations' locally and then 'eb deploy' and then it would've taken care of actually running the migration for me?

What I actually did was ssh into my EB host, run makemigrations and then migrate manually. Then I copied the migration file back to my local which I then committed. 

Unfortunately, I couldn't really figure out how to run my local server (which I created a separate issue for - https://github.com/onnela-lab/beiwe-backend/issues/112)

**Comment by biblicabeebli** [comment]
No, the migrations are created by developers and are committed to the repo. You do not have to create them.  Running makemigrations off of the master or production branches will not (well, okay, _should_ not) create new migrations.  Migrations are run, when necessary, on every deploy operation.  

The only thing I can think of is that you deleted migration files or removed/edited one of the elastic beanstalk config files (`.ebextensions/01.config`).

reviewing your other issue.

**Comment by ravibanks54** [comment]
Ah I see, in my situation, I was attempting to make migrations since I was adding new sensor settings in the backend. Had no idea that there's a hook in .ebextensions that will auto-run migrate during deploy.

I'll write up a wiki page on the following
1. Setting up beiwe locally
2. Updating sensor settings
3. Migrating the correct way

**Comment by biblicabeebli** [comment]
I just posted one.
https://github.com/onnela-lab/beiwe-backend/wiki/How-do-I-run-Beiwe-locally%3F

Can you go off of it and provide more detail?

**Comment by biblicabeebli** [comment]
We are trying to work out how to allow non-team members access to the wiki, I'm missing something because it seems like it should be a checkbox, but ... it isn't.

I'm just going to give you edit access for now, please do not destroy the repo.  :b

**Comment by ravibanks54** [comment]
Thanks Eli! I'll update it once I get some free time. And yep for sure, will definitely not mess with the repo other than the wiki. (It's also really weird that GitHub doesn't have finer grained permissions)

**Comment by biblicabeebli** [comment]
I think it is an ambiguously places wording issue, I need to make a fake account and fiddle with granting it permissions.
